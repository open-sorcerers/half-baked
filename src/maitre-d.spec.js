// import { flexeca } from 'flexeca'
// import execa from 'execa'
import why from 'why-is-node-running'
import axios from 'axios'
import { fork } from 'fluture'
import maitreD from './index'

describe("servers, who needs 'em?", () => {
  let saved
  const sockets = []

  const bind = (d) => ({ app, state }) => {
    // server = app.listen(2345, d)
    state.server.on('connection', (socket) => {
      sockets.push(socket)
    })
    saved = state
    d()
  }

  const masterServer = maitreD({
    PORT: 2345,
    logging: false
  })

  beforeAll((done) => {
    fork(done)(bind(done))(masterServer)
  })
  afterAll((done) => {
    if (saved && saved.server) {
      saved.server.removeAllListeners()
      saved.server.unref()
      saved.server.close(() => {
        saved.onUnload()
        done()
        // process.exit(0) // or `--forceExit`
      })
    }
  })

  test('GET /', (done) => {
    return axios({ method: 'get', url: 'http://localhost:2345' })
      .catch(done)
      .then((out) => {
        expect(out.data).toMatchSnapshot()
        // stop()
        done()
      })
  })
})
