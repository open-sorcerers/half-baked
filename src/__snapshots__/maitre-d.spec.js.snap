// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`servers, who needs 'em?  GET / 1`] = `
Array [
  Object {
    "id": "cool",
    "this": "is pretty cool",
  },
]
`;

exports[`servers, who needs 'em?  GET /:id - cool 1`] = `
Object {
  "data": Object {
    "id": "cool",
    "this": "is pretty cool",
  },
}
`;
